### ISSUE TITLE
Requests with invalid references produce Internal Server Error

### ISSUE THREAD

**Issue by danvk** [ga4gh/server#487]
For example, if you're trying to request alignments from a BAM file with `chr20` as the reference, not `20`, you get:

```
$ curl .../v0.5.1/reads/search --data '{"readGroupIds":["foo:bar"],"referenceName":"20","start":144700,"end":144800}' | jq .
{
  "errorCode": 3424984409,
  "message": "Internal Server Error"
}
```

You have to look at the server logs to see the real error:

```
Exception on /v0.5.1/reads/search [POST]
--------------------------------------------------------------------------------
Traceback (most recent call last):
...
  File "pysam/calignmentfile.pyx", line 641, in pysam.calignmentfile.AlignmentFile._parseRegion (pysam/calignmentfile.c:8888)
ValueError: invalid reference `20`
```

It would be better to include the real error in the response:

```
$ curl .../v0.5.1/reads/search --data '{"readGroupIds":["foo:bar"],"referenceName":"20","start":144700,"end":144800}' | jq .
{
  "errorCode": 3424984409,
  "message": "invalid reference `20`"
}
```

(somewhat related to #486)

**Comment by jeromekelleher** [comment]
This is true. @dcolligan, is there some interface in pysam to allow us to test if a reference is valid or not? We should try this before sending the request down to htslib as part of our sanitisation steps, as this will allow us to make a good error message.

**Comment by dcolligan** [comment]
Thanks for the bug report, @danvk ; I'm looking into it.  I was able to reproduce this bug with our example data with this call:
```
curl http://localhost:8000/v0.5.1/reads/search --header 'Content-Type: application/json' --data '{"readGroupIds":["low-coverage:HG00096.mapped.ILLUMINA.bwa.GBR.low_coverage.20120522"],"referenceName":"chr20","start":144700,"end":144800}'
```

**Comment by dcolligan** [comment]
A fix has been merged into the `release-0.1` branch and will be included in an upcoming release.

**Comment by dcolligan** [comment]
We've released a new version; closing...
